adalflow/adalflow/components/model_client/azureai_client.py

```diff
@@ -0,0 +1,399 @@
+
+
+def handle_streaming_response(generator: Stream[ChatCompletionChunk]):
+    r"""Handle the streaming response."""
+    for completion in generator:
+        log.debug(f"Raw chunk completion: {completion}")
+        parsed_content = parse_stream_response(completion)
+        yield parsed_content
+
+
+def get_all_messages_content(completion: ChatCompletion) -> List[str]:
+    r"""When the n > 1, get all the messages content."""
+    return [c.message.content for c in completion.choices]
+
+
+def get_probabilities(completion: ChatCompletion) -> List[List[TokenLogProb]]:
+    r"""Get the probabilities of each token in the completion."""
+    log_probs = []
+    for c in completion.choices:
+        content = c.logprobs.content
+        print(content)
+        log_probs_for_choice = []
+        for openai_token_logprob in content:
+            token = openai_token_logprob.token
+            logprob = openai_token_logprob.logprob
+            log_probs_for_choice.append(TokenLogProb(token=token, logprob=logprob))
+        log_probs.append(log_probs_for_choice)
+    return log_probs
+
+
+class AzureAIClient(ModelClient):
+    __doc__ = r"""A component wrapper for the AzureOpenAI API client.
+
+    Support both embedding and chat completion API.
+
+    Users (1) simplify use ``Embedder`` and ``Generator`` components by passing OpenAIClient() as the model_client.
+    (2) can use this as an example to create their own API client or extend this class(copying and modifing the code) in their own project.
+
+    Note:
+        We suggest users not to use `response_format` to enforce output data type or `tools` and `tool_choice`  in your model_kwargs when calling the API.
+        We do not know how AzureOpenAI is doing the formating or what prompt they have added.
+        Instead
+        - use :ref:`OutputParser<components-output_parsers>` for response parsing and formating.
+
+    Args:
+        api_key (Optional[str], optional): AzureOpenAI API key. Defaults to None.
+        chat_completion_parser (Callable[[Completion], Any], optional): A function to parse the chat completion to a str. Defaults to None.
+            Default is `get_first_message_content`.
+
+    References:
+        - Embeddings models: https://platform.openai.com/docs/guides/embeddings
+        - Chat models: https://platform.openai.com/docs/guides/text-generation
+        - AzureOpenAI docs: https://learn.microsoft.com/en-us/azure/ai-services/openai/overview
+    """
+
+    def __init__(
+        self,
+        api_key: Optional[str] = None,
+        api_version:Optional[str]=None,
+        azure_endpoint: Optional[str]= None,
+        credential: Optional[DefaultAzureCredential] = None,
+        chat_completion_parser: Callable[[Completion], Any] = None,
+        input_type: Literal["text", "messages"] = "text",
+    ):
+        r"""It is recommended to set the OPENAI_API_KEY environment variable instead of passing it as an argument.
```

Comment:
so it does not have it only api_key? why using openai_api_key?

Reply:
@liyin2015 I have made changes as mentioned in the comments